packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/CoreTests.gen.kt

```diff
@@ -179,7 +179,7 @@ data class TestMessage (
   companion object {
     @Suppress("UNCHECKED_CAST")
     fun fromList(list: List<Any?>): TestMessage {
-      val testList = list[0] as? List<Any?>
+      val testList = list[0] as List<Any?>?
```

Comment:
The kotlin version is still `List<Any?>`, but Swift changed from `[Any?]` to `[Any]`. Is it intended?

Reply:
Swift optional Any causes issues with force casting into optional types when the data is nil. Kotlin doesn't have that problem.

Reply:
> Swift optional Any causes issues with force casting into optional types when the data is nil.

Not sure if I follow this. Can you share a short code snippet when it fails? Are you talking about this following scenario? 

```
let list: [Any?] = [nil, 1, "Hi"]
let optionalBool = list[0] as! Bool?
```

Reply:
this is the scenario that was happening before. The only difference now is 
`list: [Any]`

Reply:
Interesting. This scenario actually did not crash when I try it on the playground.

Reply:
> The kotlin version is still `List<Any?>`, but Swift changed from `[Any?]` to `[Any]`. Is it intended?

The core answer here is that `Any` is fundamentally different between the two languages: `Any` is a root type in Swift, but not Kotlin.

This Kotlin code:
```
val i: Int? = null
println(i is Any)
println(i is Any?)
```

prints
```
false
true
```

but this Swift code:
```
var i: Int? = nil
print(i is Any)
print(i is Any?)
```

prints
```
true
true
```

We're generating different code because the expression of "any type, including nullable types" is different in the two languages.

(I wasn't aware of this difference when reviewing the initial generator code, so didn't catch the incorrect use of `Any?` in Swift in the first place.)